x-pack/plugins/maps/public/routing/routes/maps_app/maps_app_view.js

```diff
@@ -237,18 +245,10 @@ export class MapsAppView extends React.Component {
     );
   }
 
-  _onTopNavRefreshConfig = ({ isPaused, refreshInterval }) => {
-    this._onRefreshConfigChange({
-      isPaused,
-      interval: refreshInterval,
-    });
-  };
+  _updateStateFromSavedQuery = (savedQuery) => {
+    this.setState({ savedQuery: { ...savedQuery } });
+    this._appStateManager.setQueryAndFilters({ savedQuery });
 
-  _updateStateFromSavedQuery(savedQuery) {
-    if (!savedQuery) {
```

Comment:
I want to say I added this condition after triggering it somehow but I can't quite remember how. Are you certain this should be removed?

Reply:
Also, what's the point of executing the remaining logic if there's no saved query?

Reply:
TLDR: `_updateStateFromSavedQuery` is no longer called with `undefined`

details:

Prior to these changes, `_updateStateFromSavedQuery` was getting called with undefined from on `onClearSavedQuery`

```
  const onClearSavedQuery = () => {
    onQuerySaved(undefined);
    onQueryChange({
      filters: filterManager.getGlobalFilters(),
      query: queryString.getDefaultQuery(),
    });
  };
```

Now `onClearSavedQuery` does not call `_updateStateFromSavedQuery`. Its just

```
       onClearSavedQuery={() => {
          const { filterManager, queryString } = getData().query;
          this.setState({ savedQuery: '' });
          this._appStateManager.setQueryAndFilters({ savedQuery: '' });
          this._onQueryChange({
            filters: filterManager.getGlobalFilters(),
            query: queryString.getDefaultQuery(),
          });
        }}
```